src/main/java/iterator/RichIterator.java

```diff
@@ -452,7 +795,53 @@ default RichIterator<A> dropUntil(Predicate<? super A> predicate) {
      * @return the iterator
      */
     default BufferedIterator<A> buffered() {
-        throw new NotImplementedException();
+        return new BufferedIterator<A>() {
+            private A head;
+
+            @Override
+            public boolean hasNext() {
+                return headOptional().isPresent();
+            }
+
+            @Override
+            public A next() {
+                Optional<A> element = headOptional();
+                Optional<A> newHead = RichIterator.this.nextOptional();
+
+                head = newHead.orElse(null);
+
+                if (element.isPresent()) {
+                    return element.get();
+                } else {
+                    throw new NoSuchElementException();
+                }
+            }
+
+            @Override
+            public A head() {
+                if (head == null && RichIterator.this.hasNext()) {
+                    head = RichIterator.this.next();
+                }
+
+                if (head == null) {
+                    throw new NoSuchElementException();
+                }
+
+                return head;
+            }
+
+            @Override
+            public Optional<A> headOptional() {
+                A element;
+                try {
+                    element = head();
+                } catch (NoSuchElementException exception) {
+                    return Optional.empty();
+                }
```

Comment:
This would not pass tests...

Avoid throwing Runtime Expcetions (classes that inherit from `RuntimeException`) if it can be avoided (by definition it can).